### PR TITLE
GCI PRs for opt-common-subexpr

### PR DESCRIPTION
**Pull request by saxenism**

### CODE REVIEW

**Review by pachadotdev — APPROVED** [review]
ok, it describes what is expected

File: vignettes/opt-common-subexpr.Rmd [hunk 1]
```diff
@@ -30,6 +30,10 @@ speed_up <- function(obj) {
 
 # Common Subexpression Elimination
 
+## Idea
+
+The main idea behind the common subexpression elimination optimizer is to separate/isolate those parts of an expression(sub-expressions) that are being calculated again and again. By isolating a common sub-expression and replacing it with a varialbe, we can limit the number of computations. Thus improving the performance of our script
```
Comment (jcrodriguez1989) [inline vignettes/opt-common-subexpr.Rmd]: Please add a space in `expression(sub-expressions)`.

File: vignettes/opt-common-subexpr.Rmd [hunk 2]
```diff
@@ -30,6 +30,10 @@ speed_up <- function(obj) {
 
 # Common Subexpression Elimination
 
+## Idea
+
+The main idea behind the common subexpression elimination optimizer is to separate/isolate those parts of an expression (sub-expressions) that are being calculated again and again. By isolating a common sub-expression and replacing it with a varialbe, we can limit the number of computations. Thus improving the performance of our script
```
Comment (jcrodriguez1989) [inline vignettes/opt-common-subexpr.Rmd]: Please add an ending dot for the paragraph.

Comment (jcrodriguez1989) [reply vignettes/opt-common-subexpr.Rmd]: typo: `varialbe` ~> `variable`

File: R/opt-common-subexpr.R [hunk 3]
```diff
@@ -16,6 +16,13 @@
 #'   sep = "\n"
 #' )
 #' cat(opt_common_subexpr(list(code))$codes[[1]])
+#' 
+#' heron_formula <- paste(
+#'   "area <- (a/2 + b/2 +c/2) * (a/2 + b/2 + c/2 - a) * (a/2 + b/2 + c/2 - b) * (a/2 + b/2 + c/2 - c)",
```
Comment (jcrodriguez1989) [inline R/opt-common-subexpr.R]: this line gets too long (> 80 spaces), break it as:
```
  "area <- (a/2 + b/2 +c/2) * (a/2 + b/2 + c/2 - a) * (a/2 + b/2 + c/2 - b) *",
  "  (a/2 + b/2 + c/2 - c)",
```

**Pull request merged**